### PR TITLE
Refactor ES6 challenge descriptions and titles

### PR DESCRIPTION
**Pull request by robbawebba**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX
Progress towards #12698
#### Description

reformat ES6 challenge descriptions, titles, and seed code where necessary.

### CODE REVIEW

**Comment by Greenheart** [comment]
@robbawebba Seems like you've got a minor merge conflict on one of the titles. If you resolve it, I'll review this :blush:

**Comment by Greenheart** [comment]
@robbawebba Oops, looks like you did a `git pull` instead of `git pull --rebase`.

But now it's fixed?

**Comment by robbawebba** [comment]
@Greenheart thanks for the heads up! Resolved :)

**Comment by robbawebba** [comment]
@Greenheart whoops, I didn't see your 2nd comment - Yeah, I goofed up a little bit with my merging the first time, but it's all good now! 😅

**Review by Greenheart — CHANGES_REQUESTED** [review]
@robbawebba Great work so far!

I only found some minor issues :blush:

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (robbawebba) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: I think I may prefer to drop the word "the" altogether so it reads "Use template literal syntax" because it sounds cleaner. What do you think?

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (robbawebba) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: I changed "first" to "firstly" on line 547 as well for consistency 🙂

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (robbawebba) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: whoops, apologies for not thinking about that sooner! 😅 I think I may omit this altogether because there is so much variety between keyboards it would be tough to cover them all.

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (Greenheart) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: Good idea!

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (Greenheart) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: This time, I found something that might be misinterpreted:
```diff
-With ES6, You can remove the <code>function</code> keyword and colon altogether when defining functions
+With ES6, You can remove the <code>function</code> keyword and colon altogether when defining functions in objects.
```

**Review by Greenheart — CHANGES_REQUESTED** [review]
Just one more thing ;)

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json
Comment (robbawebba) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: Good catch! I agree, it's better to be more specific here.

**Comment by QuincyLarson** [comment]
@robbawebba just let me know once you've made those changes and we can merge this.

**Comment by robbawebba** [comment]
@QuincyLarson @Greenheart Just added the last suggestion, sorry for the delay!

**Comment by Greenheart** [comment]
@robbawebba No worries, I'm just as busy myself... :smile:

I don't have a local instance of fCC at the moment so I can't test this, so you'll need to find someone else to do the final QA.

**Comment by BerkeleyTrue** [comment]
@Greenheart This PR still needs you approval. Are you satisfied with the changes? Is so can you hit approve?

@freeCodeCamp/moderators Anyone have some free cycles to test this one locally?

**Review by Greenheart — APPROVED** [review]

**Comment by QuincyLarson** [comment]
@robbawebba Thanks for these improvements. These all look good to me. 

@Greenheart thanks for QA'ing!

**Pull request merged**
